### PR TITLE
Configurable smtputf8

### PR DESCRIPTION
**Pull request by gene-hightower**
Fixes #

Changes proposed in this pull request:
- 
- 

Checklist:
- [x] docs updated
- [ ] tests updated
- [x] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: connection.js [hunk 1]
```diff
@@ -918,9 +918,12 @@ class Connection {
                     `${this.local.host} Hello ${this.get_remote('host')}${this.ehlo_hello_message}`,
                     "PIPELINING",
                     "8BITMIME",
-                    "SMTPUTF8",
                 ];
 
+                if (!!this.cfg.main.smtputf8) {
```
Comment (msimerson) [inline connection.js:923]:
```suggestion
                if (this.cfg.main.smtputf8) {
```

**Review by msimerson — CHANGES_REQUESTED** [review]
Other than the lint failure due to "Redundant double negation", the implementation looks good. Please fix that and also add an entry to the Changes.md file. 

I'm also curious why you'd want to disable SMTPUTF8.

File: connection.js (hunk 1 above)
Comment (gene-hightower) [reply connection.js:923]: I'm new to JavaScript, won't this throw is this.cfg.main.smtputf8 is undef?

**Comment by gene-hightower** [comment]
> I'm also curious why you'd want to disable SMTPUTF8.

In this particular setup I'm working on Haraka is not the final hop in the chain, and subsequent hops may or may not support the SMTPUTF8 extension.

File: connection.js (hunk 1 above)
Comment (msimerson) [reply connection.js:923]: Nope. JS isn't strongly typed. In a case like this, undefined is the same as "not true."

**Review by msimerson — DISMISSED** [review]
Looks good. If you'll also update the Changes.md file, that'd be great.

```diff
diff --git a/Changes.md b/Changes.md
index 6398821c..a7df7637 100644
--- a/Changes.md
+++ b/Changes.md
@@ -1,6 +1,14 @@
 
 ## NEXT - 2020-MM-DD
 
+### Changes
+
+### New features
+
+- add ability to disable SMTPUTF8 advertisement #2866
+
+### Fixes
+
 
 ## 2.8.26 - 2020-11-18
 
```

**Review by msimerson — APPROVED** [review]

File: docs/CoreConfig.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ The list of plugins to load
     specify -1 to disable spooling completely or 0 to force all messages to be spooled to disk.
   * graceful\_shutdown - (default: false) enable this to wait for sockets on shutdown instead of closing them quickly
   * force_shutdown_timeout - (default: 30) number of seconds to wait for a graceful shutdown
+  * smtputf8 - (default: true) advertise support for SMTPUTF8
```
Comment (spathon) [inline docs/CoreConfig.md:44]: Isn't it default false?

Comment (msimerson) [reply docs/CoreConfig.md:44]: The default is enabled. See [here](https://github.com/haraka/Haraka/blob/master/server.js#L37)

Comment (spathon) [reply docs/CoreConfig.md:44]: Ah thanks :)

**Pull request merged**